### PR TITLE
Fix listing management server by parameters

### PR DESCRIPTION
**Pull request by davidjumani**
## Description
The List Management Server api returns a list of all the management servers but fails when trying to list by id or name. This ensures that it fetches the details as per the parameters passed.
Fixes: #3833

## Types of changes
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
Verified by testing the api with and without parameters

### CODE REVIEW

**Comment by rhtyd** [comment]
@davidjumani can you rebase against 4.13, as the issue asked for PR to be against 4.13 branch?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
LGTM based on code changes

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan help

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 1]
```diff
@@ -3832,20 +3832,43 @@ protected ResourceDetailResponse createResourceDetailsResponse(ResourceDetail re
...
-            mgmtResponse.setState(mgmt.getState());
-            mgmtResponse.setVersion(mgmt.getVersion());
-            mgmtResponse.setObjectName("managementserver");
-            result.add(mgmtResponse);
-        }
-        response.setResponses(result);
+        Pair<List<ManagementServerHostVO>, Integer> result = listManagementServersInternal(cmd);
+        List<ManagementServerResponse> hostResponses = new ArrayList<>();
+
+        for(ManagementServerHostVO host : result.first()) {
```
Comment (rhtyd) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java]: Add space after for and ifs

**Pull request merged**